mono/metadata/threads.c

```diff
@@ -1634,7 +1634,8 @@ gint32 ves_icall_System_Threading_WaitHandle_WaitAll_internal(MonoArray *mono_ha
 
 	g_free(handles);
 
-	return ret;
+	/* WAIT_FAILED in waithandle.cs is different from WAIT_FAILED in Win32 API */
```

Comment:
WAIT_FAILED in Win32 is defined as a DWORD 0xFFFFFFFF, i.e. a 32bit _unsigned_ integer.
The constant in waithandle.cs is defined as a C# int 0x7FFFFFFF, i.e. a 32bit _signed_ integer.

Since we're converting from unsigned to signed here, isn't this undefined behavior? We'd be hitting the same case everywhere where WAIT_FAILED could be returned, e.g. WaitForSingleObjectEx in mono_wait_uninterrupted().